### PR TITLE
lbmap: Initialize maps before test suite runs

### PR DESCRIPTION
**Pull request by christarazi**
This commit adds the missing initialization needed for BPF maps à la
d9fa628ea ("daemon, lbmap: Avoid premature init of BPF maps"). This
fixes the following test failure in the privileged tests:

```
$ sudo -E make -j $(nproc) TESTPKGS=pkg/maps/lbmap tests-privileged
...
=== RUN   Test
START: maglev_test.go:43: MaglevSuite.SetUpSuite
PASS: maglev_test.go:43: MaglevSuite.SetUpSuite 0.000s

START: maglev_test.go:73: MaglevSuite.TestInitMaps
level=info msg="Deleting Maglev outer map due to different M or empty map" bpfMapName=cilium_lb4_maglev subsys=map-lb
level=info msg="Deleting Maglev outer map due to different M or empty map" bpfMapName=cilium_lb4_maglev subsys=map-lb
level=info msg="Deleting Maglev outer map due to different M or empty map" bpfMapName=cilium_lb4_maglev subsys=map-lb
... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0x4399D8)

/usr/lib/go/src/runtime/panic.go:969
  in gopanic
/usr/lib/go/src/runtime/panic.go:212
  in panicmem
/usr/lib/go/src/runtime/signal_unix.go:742
  in sigpanic
/home/chris/code/cilium/cilium/pkg/bpf/map_linux.go:464
  in Map.OpenOrCreate
lbmap.go:551
  in updateServiceEndpoint
lbmap.go:119
  in LBBPFMap.UpsertService
maglev_test.go:103
  in MaglevSuite.TestInitMaps
/usr/lib/go/src/reflect/value.go:337
  in Value.Call
/usr/lib/go/src/runtime/asm_amd64.s:1374
  in goexit
PANIC: maglev_test.go:73: MaglevSuite.TestInitMaps

OOPS: 0 passed, 1 PANICKED
--- FAIL: Test (0.11s)
FAIL
coverage: 10.4% of statements in github.com/cilium/cilium/pkg/maps/lbmap
FAIL    github.com/cilium/cilium/pkg/maps/lbmap 0.123s
FAIL
make: *** [Makefile:146: tests-privileged] Error 1
```

Fixes: d9fa628ea

Signed-off-by: Chris Tarazi <chris@isovalent.com>

### CODE REVIEW

**Comment by christarazi** [comment]
retest-runtime

**Review by Weil0ng — APPROVED** [review]
Thanks Chris!

File: pkg/maps/lbmap/maglev_test.go
Comment (aditighag) [inline pkg/maps/lbmap/maglev_test.go:67]: Do we want to move the initialization of maps to `SetupTest` in case the individual test cases manipulate maps?

**Review by aditighag — APPROVED** [review]

File: pkg/maps/lbmap/maglev_test.go
Comment (christarazi) [reply pkg/maps/lbmap/maglev_test.go:67]: No, I don't think so because the previous behavior would be that the map objects are initialized by virtue of being package-level variables, in other words, they'd be initialized once. Putting this call here now has the same effect of initializing the objects once.

Plus, I don't think the objects need to be reinitialized for each test run also. The contents aren't changing or anything like that.

**Comment by brb** [comment]
Why we didn't see this failure in the CI? Because the maps were initialized by other test suite?

**Review by brb — APPROVED** [review]

**Comment by christarazi** [comment]
> Why we didn't see this failure in the CI? Because the maps were initialized by other test suite?

Not sure, I was wondering the same. There was discussion about it here: https://cilium.slack.com/archives/C7PE7V806/p1612302871017900

**Comment by christarazi** [comment]
At this point, we have code owner approval and we only needed the runtime tests to pass. Marking as ready to merge.

**Review by jibi — APPROVED** [review]

**Pull request merged**
